source/common/network/udp_listener_impl.cc

```diff
@@ -0,0 +1,163 @@
+#include "common/network/udp_listener_impl.h"
+
+#include <sys/un.h>
+
+#include <cassert>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/exception.h"
+
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/fmt.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/network/address_impl.h"
+
+#include "event2/listener.h"
+
+namespace Envoy {
+namespace Network {
+
+UdpListenerImpl::UdpListenerImpl(const Event::DispatcherImpl& dispatcher, Socket& socket,
+                                 UdpListenerCallbacks& cb)
+    : BaseListenerImpl(dispatcher, socket), cb_(cb) {
+  event_assign(&raw_event_, &dispatcher.base(), socket.ioHandle().fd(),
```

Comment:
Any reason to not just use a FileEvent here?

Reply:
I could have used `FileEvent` but was not familiar with the API, especially how the enable/disable functionality would work.

Reply:
You just call `setEnabled()`, which IIRC has no events specified will disable everything. See https://github.com/envoyproxy/envoy/blob/master/include/envoy/event/file_event.h#L47. It's a pretty simple interface/impl. I think we should use a FileEvent under the hood to avoid duplicating the logic here and more easily allow us to move away from libevent if needed in the future.

Reply:
I have updated the PR with the change suggested.